lib/strings.nix

```diff
@@ -85,7 +94,7 @@ rec {
     list:
     if list == [] || length list == 1
     then list
-    else tail (lib.concatMap (x: [separator x]) list);
+    else builtins.tail (lib.concatMap (x: [separator x]) list);
```

Comment:
Instead of doing this, you can add
```nix
  inherit (lib) head tail;
```

to the top `let in`

(the preference should be to get symbols from `lib`, because it's more flexible)

Reply:
The problem with that is the top-level `let-in` is above the `rec { }` attrset where we define the deprecated `head` and `tail` with warnings... so any unqualified references will refer to those and throw warnings.

(That's why at first I had pulled those references out into a separate attrset that was merged in with `//` before...)

I can access them from `lib` instead of `builtins` though if that's preferred

Reply:
Moved to using `lib.{head,tail}` instead of `builtins.{head,tail}` as requested. As mentioned I don't see how a `let-inherit` would help here since the `head`/`tail` with warnings in the returned recursive attrset would shadow them. Please let me know if I'm missing something, or if there's a different approach you would prefer...